.github/workflows/run_tests.yml

```diff
@@ -28,12 +44,33 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          python -m pip install -r ./tests/requirements.txt
-
-      - name: Checks with pre-commit
-        uses: pre-commit/action@v2.0.3
+          python -m pip install '.[test]'
 
       - name: Test with pytest
         run: |
-          python -m pip install .
           python -m pytest
+
+  oldest-supported:
+    needs: [format]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          enable-cache: true
+          cache-dependency-glob: "**/pyproject.toml"
+
+      - name: "Set up Python"
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Install the project
+        run: uv sync --extra test --resolution lowest
```

Comment:
Q: When using `uv` I've typically used `--resolution lowest-direct`, because typically there is some transitive dependency which hasn't correctly specified its lower bounds. No strong feelings on it here -- we have almost zero dependencies anyway -- but I'm just curious about your experience on this one?

Reply:
Yeah, I've used both and found ``--resolution lowest-direct`` to be problematic.
The issue is that `lowest-direct` only does the earliest version of the direct dependency but then the most-up-to-date version of all other dependencies, which is not a realistic reflection of a user's environment. ```--resolution lowest`` is much more realistic. It does force the slightly annoying requirement of setting constraints in uv (as done here) when dependencies don't have proper minimum version pins (e.g. Jupyter is bad at this), but for me this has repeatedly proven useful in diagnosing problems.